src/ivoc/ivocrand.cpp

```diff
@@ -183,7 +184,7 @@ void RandomPlay::play() {
     *px_ = (*(r_->rand))();
 }
 void RandomPlay::list_remove() {
-    for (auto it = random_play_list_->begin(); it != random_play_list_->end(); ++it) {
+    for (auto it: iterate(*random_play_list_)) {
```

Comment:
Many operations on iterators cause invalidation. Frequently `end()`. This matters because range based for loops expand to something akin to:
```
{

    auto && __range = range-expression ;
    auto __begin = begin-expr ;
    auto __end = end-expr ;
    for ( ; __begin != __end; ++__begin)
    {

        range-declaration = *__begin;
        loop-statement

    }

}
```
https://en.cppreference.com/w/cpp/language/range-for

Note that the `end-expr` is evaluated once; not once per loop. Hence, if anything invalidates the end, e.g. by erasing, inserting elements past the capacity, etc. then the code can't use `iterate`. The question is when does one need an iterator, but doesn't need to perform operation that might invalidate any of the involved iterators.

Reply:
`iterate` has been removed, it was a bad idea because of this reason